editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,357 @@
+        // if the feature is a point, move the point
+        movePoint(mapPoint);
+      }
+    }
+  }
+
+  /**
+   * Identifies a feature near the given screen point.
+   *
+   * @param screenPoint location the user clicked
+   */
+  private void identifyFeature(Point2D screenPoint) {
+
+    // clear any previously selected features
+    clearSelection();
+
+    // identify across all layers
+    ListenableFuture<List<IdentifyLayerResult>> identifyLayerResultsFuture = mapView.identifyLayersAsync(screenPoint, 1, false);
+    identifyLayerResultsFuture.addDoneListener(() -> {
+      try {
+        // get the list of results from the future
+        List<IdentifyLayerResult> identifyLayerResults = identifyLayerResultsFuture.get();
+        // if one or more results have been identified
+        if (!identifyLayerResults.isEmpty()) {
+          // retrieve the first result
+          IdentifyLayerResult layerResult = identifyLayerResults.get(0);
+          // check that the result is a feature layer, thereby excluding annotation layers
+          if (layerResult.getLayerContent() instanceof FeatureLayer) {
+            // get a reference to the identified feature
+            selectedFeature = (Feature) layerResult.getElements().get(0);
+            // check the geometry and select the feature
+            selectFeature(layerResult);
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    });
+  }
+
+  /**
+   * Checks if the identified feature is a straight polyline or a point, and select the feature.
+   * For a point feature, show a dialog to edit attributes. Future clicks will call move functions.
+   *
+   * @param layerResult identify layer result
+   */
+  private void selectFeature(IdentifyLayerResult layerResult) {
+
+    try {
+      // if the selected feature is a polyline
+      if (selectedFeature.getGeometry().getGeometryType() == GeometryType.POLYLINE) {
+        // create a polyline builder from the selected feature
+        PolylineBuilder polylineBuilder = new PolylineBuilder((Polyline) selectedFeature.getGeometry());
+        // get a list of parts of the selected polyline
+        List<Part> parts = polylineBuilder.getParts();
+        parts.forEach(part -> {
+          // if the selected feature is a polyline with any part containing more than one segment
+          // (i.e. a curve)
+          if (part.getPointCount() > 2) {
+            selectedFeature = null;
+            // show message reminding user to select straight (single segment) polylines only
+            new Alert(Alert.AlertType.WARNING, "Select straight (single segment) polylines only.").show();
+            // return early, effectively disallowing selection of multi segmented polylines
+            return;
+          } else {
+            // select the polyline feature
+            ((FeatureLayer) layerResult.getLayerContent()).selectFeature(selectedFeature);
+            selectedFeatureIsPolyline = true;
+          }
```

Comment:
I personally think we can simplify this by turning the if-statment around. Doing an 'early return' is very much a C# pattern, as far as I'm aware.

```suggestion
          // only select single segment lines
          if (part.getPointCount() <= 2) {
            // select the polyline feature
            ((FeatureLayer) layerResult.getLayerContent()).selectFeature(selectedFeature);
            selectedFeatureIsPolyline = true;
          
          } else {
            selectedFeature = null;
            // show message reminding user to select straight (single segment) polylines only
            new Alert(Alert.AlertType.WARNING, "Select straight (single segment) polylines only.").show();
          }
```